generator/lib/control.ml

```diff
@@ -46,6 +46,35 @@ let api =
           ];
         ]
       )) in
+  let configuration = {
+    Arg.name = "configuration";
+    ty = Type.(Dict(String, Basic String));
+    description = String.concat " " [
+                    "Plugin-specific configuration which describes where and";
+                    "how to locate the storage repository. This may include";
+                    "the physical block device name, a remote NFS server and";
+                    "path or an RBD storage pool.";
+                  ];
+  } in
+  let probe_result_decl =
+    Type.(Struct (
+        ( configuration.name, configuration.ty, configuration.description ),
+        [ "complete", Basic Boolean, String.concat " " [
+            "True if this configuration is complete and can be used to";
+            "call SR.create. False if it requires further iterative calls";
+            "to SR.probe, to potentially narrow down on a configuration";
+            "that can be used.";
```

Comment:
And what happens if there is an existing `sr` with this configuration? (when `sr` is `Some ...`)

Reply:
You can also call SR.attach, pushed a fixup